File: 10_StructuralTyping/c#/Program.cs
namespace Demo
{
    // logger interface
    interface ILogger
    {
        void Log(string message);
    }

    // logger implementation that uses Console
    class ConsoleLogger : ILogger
    {
        public void Log(string message)
        {
            System.Console.WriteLine(message);
        }
    }

    class Program
    {
        public static void Main(string[] args)
        {
            ILogger logger;
            logger = new ConsoleLogger();
            logger.Log("Hello");
            logger.Log("World");
        }
    }
}




































//     class NotAConsoleLogger
//     {
//     	public void Log(string message)
//         {
//             System.Console.WriteLine(message);
//         }
//     }

File: 10_StructuralTyping/java/Program.java

interface ILogger {
    void log(String message);
}

class ConsoleLogger implements ILogger {
    public void log(String message) {
        System.out.println(message);
    }
}

class Program {

    public static void main(String[] args) {
        ILogger logger = new ConsoleLogger();

        logger.log("Hello");
    }
}